### PR TITLE
Add more checks to Lint

### PR DESCRIPTION
**Pull request by Sam1301**
I think while reviewing prs raising errors for unused namespaces, unused attributes and use of deprecated resources will be helpful.

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@Sam1301, it looks like you've already signed the Dropbox CLA. Thanks!

**Comment by timabbott** [comment]
@Sam1301 this looks great; I'm always a fan of tightening what computers can check for us :)

Can you fix the commit message?  The first line is longer than 76 characters.  
Check out the Zulip commit message guidelines for more details: https://zulip.readthedocs.io/en/latest/version-control.html#commit-messages

**Comment by Sam1301** [comment]
Thanks, changed the commit message :)

File: app/lint.xml [hunk 1]
```diff
@@ -13,5 +13,7 @@
         <ignore path="**/ic_person_24dp.png"/>
         <ignore path="**/ic_search_24dp.png"/>
     </issue>
-
+    <issue id="UnusedNamespace" severity="error"/>
```
Comment (kunall17) [inline app/lint.xml]: These all can be added as well 
OnClick
MissingPermission
UnknownId
DuplicateIds
SdCardPath
NewerVersionAvailable
RequiredSize
Typos


Warnings:
PluralsCandidate
StaticFieldLeak

Comment (Sam1301) [inline app/lint.xml]: Thanks I included SdCardPath, NewerVersionAvailable and Typos issues. For others I think lint probably already raises errors on OnClick,MissingPermission,UnknownId,DuplicateIds and RequiredSize and warnings on PluralsCandidate, StaticFieldLeak.

Comment (kunall17) [inline app/lint.xml]: > probably already raises errors on 

You sure?

Comment (Sam1301) [inline app/lint.xml]: yes, this is the [list](http://tools.android.com/tips/lint-checks) of checks android lint already runs when we execute ./gradlew lint  or ./gradlew lintDebug
In lint.xml we could change the severity of these checks, hence changed the severity of  SdCardPath, NewerVersionAvailable and Typos issues from warnings to errors. Please correct me if I am wrong :)

**Comment by kunall17** [comment]
@timabbott LGTM!

**Comment by kunall17** [comment]
@timabbott LGTM!

**Comment by timabbott** [comment]
Oops, missed this yesterday, merged, thanks @Sam1301 and @kunall17!  Hopefully this will save a lot of time :)

**Pull request closed without merging**